Makefile

```diff
@@ -1,62 +1,61 @@
 
-# dev creates binaries for testing Vault locally. These are put
-# into ./bin/ as well as $GOPATH/bin, except for quickdev which
-# is only put into /bin/
-quickdev: generate
-	@CGO_ENABLED=0 go build -i -tags='$(BUILD_TAGS)' -o bin/vault-plugin-auth-kubernetes
-dev: fmtcheck generate
-	@CGO_ENABLED=0 BUILD_TAGS='$(BUILD_TAGS)' VAULT_DEV_BUILD=1 sh -c "'$(CURDIR)/scripts/build.sh'"
-dev-dynamic: generate
-	@CGO_ENABLED=1 BUILD_TAGS='$(BUILD_TAGS)' VAULT_DEV_BUILD=1 sh -c "'$(CURDIR)/scripts/build.sh'"
-
-# test runs the unit tests and vets the code
-test: fmtcheck generate
-	CGO_ENABLED=0 VAULT_TOKEN= VAULT_ACC= go test -tags='$(BUILD_TAGS)' $(TEST) $(TESTARGS) -timeout=20m -parallel=4
-
-testcompile: fmtcheck generate
-	@for pkg in $(TEST) ; do \
-		go test -v -c -tags='$(BUILD_TAGS)' $$pkg -parallel=4 ; \
-	done
-
-# testacc runs acceptance tests
-testacc: fmtcheck generate
-	@if [ "$(TEST)" = "./..." ]; then \
-		echo "ERROR: Set TEST to a specific package"; \
-		exit 1; \
-	fi
-	VAULT_ACC=1 go test -tags='$(BUILD_TAGS)' $(TEST) -v $(TESTARGS) -timeout 45m
-
-# generate runs `go generate` to build the dynamically generated
-# source files.
-generate:
-	go generate $(go list ./...)
-
-# bootstrap the build by downloading additional tools
-bootstrap:
-	@for tool in  $(EXTERNAL_TOOLS) ; do \
-		echo "Installing/Updating $$tool" ; \
-		go get -u $$tool; \
-	done
+.PHONY: dev
+dev:
+	CGO_ENABLED=0 go build -o bin/vault-plugin-auth-kubernetes cmd/vault-plugin-auth-kubernetes/main.go
+
+.PHONY: test
+test: fmtcheck
+	CGO_ENABLED=0 go test ./... $(TESTARGS) -timeout=20m
+
+.PHONY: integration-test
+integration-test:
+	INTEGRATION_TESTS=true CGO_ENABLED=0 go test github.com/hashicorp/vault-plugin-auth-kubernetes/integrationtest/... $(TESTARGS) -timeout=20m
 
+.PHONY: fmtcheck
 fmtcheck:
 	@sh -c "'$(CURDIR)/scripts/gofmtcheck.sh'"
 
+.PHONY: fmt
 fmt:
-	gofmt -w $(GOFMT_FILES)
-
-
-.PHONY: bin default generate test vet bootstrap fmt fmtcheck
+	gofmt -w .
+
+.PHONY: setup-kind
+# create a kind cluster for running the acceptance tests locally
+setup-kind:
+	kind get clusters | grep --silent "^${KIND_CLUSTER_NAME}$$" || \
+	kind create cluster \
+		--image kindest/node:${KIND_K8S_VERSION} \
+		--name ${KIND_CLUSTER_NAME}  \
+		--config $(CURDIR)/integrationtest/kind/config.yaml
+	kubectl config use-context kind-${KIND_CLUSTER_NAME}
+
+.PHONY: delete-kind
+# delete the kind cluster
+delete-kind:
+	kind delete cluster --name ${KIND_CLUSTER_NAME} || true
+
+.PHONY: vault-image
+vault-image:
+	GOOS=linux GOARCH=amd64 make dev
+	docker build -f integrationtest/vault/Dockerfile bin/ --tag=hashicorp/vault:dev
+
+# Create Vault inside the cluster with a locally-built version of kubernetes auth.
+.PHONY: setup-integration-test
+setup-integration-test: teardown-integration-test vault-image
+	kind --name ${KIND_CLUSTER_NAME} load docker-image hashicorp/vault:dev
+	kubectl create namespace test
+	kubectl apply --namespace=test -f integrationtest/vault/vault.yaml
```

Comment:
Is it possible to use the helm chart for deploying vault here?

Reply:
The only reason I didn't use the helm chart initially was because I need to be able to set the `hostPort` on one of the container ports (and the helm chart quite reasonably doesn't have configurability for that AFAICT). On reflection, that's a small enough change that I think helm chart + a small patch would be better for maintainability, so I've updated to do that instead.